### PR TITLE
Add View Assignment (with auto-logging in users)

### PR DESCRIPTION
**Pull request by armstrongnate**
refs: MBL-12532
affects: student
release note: none

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
#### Checkmarx scan completed with the following findings
|Lines|Severity|Category|File|Link|
---|---|---|---|---
[1](https://github.com/instructure/canvas-ios/blob/mbl-12532/AllTheThings.xcworkspace/xcshareddata/IDETemplateMacros.plist#L1) |High|App_Transport_Security_Bypass|AllTheThings.xcworkspace/xcshareddata/IDETemplateMacros.plist|[Checkmarx](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1000047&projectid=17&pathid=2)
[271](https://github.com/instructure/canvas-ios/blob/mbl-12532/Canvas/CanvasKit1/UI Classes/CKCommentViewController.m#L271) |High|Stored_XSS|Canvas/CanvasKit1/UI Classes/CKCommentViewController.m|[Checkmarx](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1000047&projectid=17&pathid=1)

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please run `yarn update-headers`. The following do not have the correct license header: 
* Canvas/CanvasUITests/Assignments/AssignmentsTests.swift
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student

[MBL-12532](https://instructure.atlassian.net/browse/MBL-12532)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Core** | 95.17% | 95.17% | 0%
**Student** | 92.89% | 92.89% | 0%
**Teacher** | 100% | 100% | 0%
**React Native** | 92.69% | 92.69% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Canvas/Canvas/CanvasAppDelegate.swift [hunk 1]
```diff
@@ -107,6 +107,7 @@ class AppDelegate: UIResponder, UIApplicationDelegate, AppEnvironmentDelegate {
 
     func setupUITests() {
         Keychain.clearEntries()
+        UIView.setAnimationsEnabled(false)
```
Comment (vanwagonet) [inline Canvas/Canvas/CanvasAppDelegate.swift]: Are we sure we want to disable animations entirely? Or would it be better to just speed them up, like we do in StudentUITests?

File: Canvas/CanvasUITests/Assignments/AssignmentsTests.swift [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+import SwiftUITest
```
Comment (vanwagonet) [inline Canvas/CanvasUITests/Assignments/AssignmentsTests.swift]: TestsFoundation

File: Canvas/CanvasUITests/CanvasUITests.swift [hunk 3]
```diff
@@ -23,20 +23,65 @@ private var getTestCase: (() -> XCTestCase)!
...
+        if let user = user {
+            application.launchArguments.append(contentsOf: [
+                "-com.apple.configuration.managed",
+                user.profile
+            ])
+        }
+        application.launch()
+        if let user = user {
+            let user = LoginStart.previousUser(name: user.username)
+            user.waitToExist(Timeout())
```
Comment (vanwagonet) [inline Canvas/CanvasUITests/CanvasUITests.swift]: no `Timeout()`, and the tap below will wait, so you don't need an explicit one here.

**Comment by vanwagonet** [comment]
Looks like a rebase or merge is in order.

**Comment by armstrongnate** [comment]
Yes yes. Waiting on a few changes from @GabrielPalmer to hit master then I will rebase.

File: Canvas/Canvas/CanvasAppDelegate.swift (hunk 1 above)
Comment (armstrongnate) [reply Canvas/Canvas/CanvasAppDelegate.swift]: Ooo I will speed them up instead.

**Comment by vanwagonet** [comment]
I don't understand why there is a compiler error for `ProfileViewController`. Apparently it wasn't a bad enough error to cause compiling to actually fail. The tests ran & passed. `Executed 6 tests, with 0 failures (0 unexpected) in 201.543 (201.549) seconds`

**Pull request merged**